spacy_llm/tasks/entity_linking.py

```diff
@@ -0,0 +1,310 @@
+import csv
+import re
+import warnings
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Type, Union
+
+import jinja2
+import spacy
+from pydantic import BaseModel
+from spacy.language import Language
+from spacy.pipeline import EntityLinker
+from spacy.scorer import Scorer
+from spacy.tokens import Doc, Span
+from spacy.training import Example
+
+from ..registry import registry
+from ..ty import CandidateSelector, ExamplesConfigType
+from .templates import read_template
+from .util import SerializableTask
+
+_DEFAULT_EL_TEMPLATE_V1 = read_template("entity_linking")
+
+
+class EntityLinkingExample(BaseModel):
+    text: str
+    mentions_str: str
+    mentions: List[str]
+    entity_descriptions: List[List[str]]
+    entity_ids: List[List[str]]
+    solutions: List[str]
+    reasons: Optional[List[str]]
+
+
+@registry.llm_misc("spacy.CandidateSelectorPipeline.v1")
+class SpaCyPipelineCandidateSelector:
+    """Callable generated by loading and wrapping a spaCy pipeline with a NEL component and a filled knowledge base."""
+
+    def __init__(
+        self,
+        nlp_path: Union[Path, str],
+        desc_path: Union[Path, str],
+        top_n: int = 5,
+    ):
+        """
+        Loads spaCy pipeline, knowledge base, entity descriptions.
+        top_n (int): Top n candidates to include in prompt.
+        nlp_path (Union[Path, str]): Path to stored spaCy pipeline.
+        desc_path (Union[Path, str]): Path to .csv file with descriptions for entities. Has to have two columns
+          with the first one being the entity ID, the second one being the description. The entity ID has to match with
+          the entity ID in the stored knowledge base.
+        """
+        self._nlp = spacy.load(nlp_path)
+        if "entity_linker" not in self._nlp.component_names:
+            raise ValueError(
+                f"'entity_linker' component has to be available in specified pipeline at {nlp_path}, but "
+                f"isn't."
+            )
+        self._entity_linker: EntityLinker = self._nlp.get_pipe("entity_linker")
+        self._kb = self._entity_linker.kb
+        with open(desc_path) as csvfile:
+            self._descs: Dict[str, str] = {}
+            for row in csv.reader(csvfile, quoting=csv.QUOTE_ALL, delimiter=";"):
+                if len(row) != 2:
+                    continue
+                self._descs[row[0]] = row[1]
```

Comment:
After this loop, I would add another check. If `self._descs` is empty, the format of the CSV was wrong and we should raise a custom warning.

Reply:
Fixed in https://github.com/explosion/spacy-llm/pull/241/commits/0e7a28d74e30ddb6950933c0227cd92df9b4af94.